source/process_common.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2010-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/process.h>
+#include <aws/common/string.h>
+
+struct aws_run_command_result *aws_run_command_result_new(struct aws_allocator *allocator) {
+    if (!allocator) {
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+        return NULL;
+    }
+    struct aws_run_command_result *result = aws_mem_calloc(allocator, 1, sizeof(struct aws_run_command_result));
+    if (!result) {
+        return NULL;
+    }
+    result->allocator = allocator;
+    return result;
+}
+
+void aws_run_command_result_destroy(struct aws_run_command_result *result) {
+    if (!result) {
+        return;
+    }
+    aws_string_destroy(result->std_out);
+    aws_string_destroy(result->std_err);
```

Comment:
aws_string_destroy_secure()?
Not sure if this is actually sensitive data or not. I just noticed you were using aws_byte_buf_clean_up_secure() to clean up the intermediate buffer in `aws_run_command()`.

Reply:
Yes, secure.